File: 003_backend/web-api/Models/Subject.cs
using System.ComponentModel.DataAnnotations;
using web_api.Models.DetailModels;

namespace web_api.Models
{
    public class Subject
    {
        [Key]
        public Guid Id { get; set; }
        public string? Name { get; set; }
        public string? ShortName { get; set; }
        public SchoolYear? SchoolYear { get; set; }

        // Listen der Klasse
        public List<Grad>? Grads { get; set; } = new List<Grad>();
    }
}

File: 002_frontend/src/app/api/models/subject.ts
/* tslint:disable */
/* eslint-disable */
import { Grad } from './grad';
import { SchoolYear } from './school-year';
export interface Subject {
  grads?: null | Array<Grad>;
  id?: string;
  name?: null | string;
  schoolYear?: SchoolYear;
  shortName?: null | string;
}
